src/HealthChecks.AzureIoTHub/AzureIoTHubtHealthCheck.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices;
+
+namespace HealthChecks.AzureIoTHub
+{
+    public class AzureIoTHubHealthCheck : IHealthCheck
+    {
+        private readonly AzureIoTHubOptions _options;
+        public AzureIoTHubHealthCheck(AzureIoTHubOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (_options.RegistryWriteCheck)
+                {
+                    await ExecuteRegistryWriteCheckAsync(cancellationToken);
+                }
+                else if (_options.RegistryReadCheck)
+                {
+                    await ExecuteRegistryReadCheckAsync();
+                }
+                if (_options.ServiceConnectionCheck)
+                {
+                    await ExecuteServiceConnectionCheckAsync(cancellationToken);
+                }
+
+                return HealthCheckResult.Healthy();
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        private async Task ExecuteServiceConnectionCheckAsync(CancellationToken cancellationToken)
+        {
+            var transportType = MapToTransportType(_options.ServiceConnectionTransport);
+            using (var client = ServiceClient.CreateFromConnectionString(_options.ConnectionString, transportType))
+            {
+                var result = await client.GetServiceStatisticsAsync(cancellationToken);
+                var _ = result.ConnectedDeviceCount;
+            }
+        }
+
+        private async Task ExecuteRegistryReadCheckAsync()
+        {
+            using (var client = RegistryManager.CreateFromConnectionString(_options.ConnectionString))
+            {
+                var query = client.CreateQuery(_options.RegistryReadQuery, 1);
+                var _ = await query.GetNextAsJsonAsync();
+            }
+        }
+
+        /// <summary>
+        /// This method try create and remove device for registry write check.
```

Comment:
can you add explanation inside methdo, typically we don't add summary comments to private members

Reply:
It is duplicate for already added in method comment. Removed